src/managed/Microsoft.Extensions.DependencyModel/DependencyContextJsonReader.cs

```diff
@@ -398,6 +398,46 @@ private List<string> ReadPropertyNames(JsonTextReader reader)
             return runtimes;
         }
 
+        private List<RuntimeFile> ReadRuntimeFiles(JsonTextReader reader)
+        {
+            var runtimeFiles = new List<RuntimeFile>();
+
+            reader.ReadStartObject();
+
+            while (reader.Read() && reader.TokenType == JsonToken.PropertyName)
+            {
+                string assemblyVersion = null;
+                string fileVersion = null;
+
+                var path = (string)reader.Value;
+
+                reader.ReadStartObject();
+
+                string propertyName;
+                string propertyValue;
+
+                while (reader.TryReadStringProperty(out propertyName, out propertyValue))
+                {
+                    if (propertyName == DependencyContextStrings.AssemblyVersionPropertyName)
+                    {
+                        assemblyVersion = propertyValue;
+                    }
+                    else if (propertyName == DependencyContextStrings.FileVersionPropertyName)
+                    {
+                        fileVersion = propertyValue;
+                    }
+                }
+
+                reader.CheckEndObject();
+
+                runtimeFiles.Add(new RuntimeFile(path, assemblyVersion, fileVersion));
+            }
+
+            reader.CheckEndObject();
+
+            return runtimeFiles;
+        }
+
         private List<RuntimeTargetEntryStub> ReadTargetLibraryRuntimeTargets(JsonTextReader reader)
```

Comment:
I think you need to update `runtimeTargets` as well.

Try building an app that references `System.Data.SqlClient/4.4.0` and you'll see what I mean:

```JSON
"System.Data.SqlClient/4.4.0": {
        "dependencies": {
          "Microsoft.Win32.Registry": "4.4.0",
          "System.Security.Principal.Windows": "4.4.0",
          "System.Text.Encoding.CodePages": "4.4.0",
          "runtime.native.System.Data.SqlClient.sni": "4.4.0"
        },
        "runtime": {
          "lib/netstandard2.0/System.Data.SqlClient.dll": {}
        },
        "runtimeTargets": {
          "runtimes/unix/lib/netstandard2.0/System.Data.SqlClient.dll": {
            "rid": "unix",
            "assetType": "runtime"
          },
          "runtimes/win/lib/netstandard2.0/System.Data.SqlClient.dll": {
            "rid": "win",
            "assetType": "runtime"
          }
        },
        "compile": {
          "ref/netstandard2.0/System.Data.SqlClient.dll": {}
        }
      },
```

Reply:
Thanks